File: UriOnline_Judge/URI_1110 - (Throwing_Cards_Away) - Accepted.cpp
#include <bits/stdc++.h>
#include <sstream>
using namespace std; 

int main(){
	int cases;
	vector<int> cola;
	//vector<int> descar;
	 
	while (cin >> cases && cases!=0){
		string salida="Discarded cards:"; 
		//int temp=0;
		//printf("Discarded cards: ");
		for(int i=cases; i>0; i--){
			cola.push_back(i);
		}
		while(cola.size()>1){
			ostringstream temp;
			temp << cola.back();
			if(cola.size()!=2){
				salida += " "+temp.str()+",";
			}else{
				salida += " "+temp.str();
			}
			//cout << salida << endl; 
			cola.pop_back();
			cola.insert(cola.begin(),cola.back());
			cola.pop_back();
		}
		cout << salida << endl; 
		cout << "Remaining card: " << cola[0] << endl;
		cola.clear();
	}
	
	
	
	return 0; 
}

File: UriOnline_Judge/URI_1110 - (Throwing Cards Away) - Accepted.java
import java.io.IOException;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

/**
 * IMPORTANT:
 *      O nome da classe deve ser "Main" para que a sua solução execute
 *      Class name must be "Main" for your solution to execute
 *      El nombre de la clase debe ser "Main" para que su solución ejecutar
 */
public class Main {

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
		while(sc.hasNext()){
			int n = sc.nextInt();
			int k;
			Queue<Integer> cola = new LinkedList<>();
			if(n!=0){
				String salida = "";
				for(int i=1; i<=n; i++) {
					cola.add(i);
				}
				while(cola.size()>1) {
					salida += String.valueOf(cola.element()) + ", ";
					cola.remove();
					k = cola.element();
					cola.remove();
					cola.add(k);
				}
				if(n>1) {
					salida = " "+salida.substring(0, salida.length()-2);
				}
				System.out.println("Discarded cards:" + salida);
				System.out.println("Remaining card: " + cola.peek());


			}else{
				System.exit(0);
				break;
			}
		}
    }

}
